mlflow/data/pandas_dataset.py

```diff
@@ -0,0 +1,128 @@
+import hashlib
+import json
+from typing import Optional, Any, Dict, Union
+
+import numpy as np
+import pandas as pd
+
+from mlflow.data.dataset import Dataset
+from mlflow.data.filesystem_dataset_source import FileSystemDatasetSource
+from mlflow.data.spark_dataset_source import SparkDatasetSource
+from mlflow.data.delta_dataset_source import DeltaDatasetSource
+
+from mlflow.data.pyfunc_dataset_mixin import PyFuncConvertibleDatasetMixin, PyFuncInputsOutputs
+from mlflow.types import Schema
+from mlflow.types.utils import _infer_schema
+
+
+class PandasDataset(Dataset, PyFuncConvertibleDatasetMixin):
+    def __init__(
+        self,
+        df: pd.DataFrame,
```

Comment:
Can we support a `targets` parameter specifying the name of the targets column? Then, in `to_pyfunc()`, can we return the targets column as a PyFuncOutput?